Transforms/table_operations/rasgo_pivot.sql

```diff
@@ -1,3 +1,13 @@
--- args: {{dimensions}}, {{list_of_vals}}, {{list_of_vals_no_quote}}, {{pivot_column}}, {{value_column}}, {{agg_method}}
+{# args: {{dimensions}}, {{pivot_column}}, {{value_column}}, {{agg_method}} #}
 
-SELECT {{dimensions}}, {{list_of_vals_no_quote}} FROM ( select {{dimensions}}, {{pivot_column}}, {{value_column}} from {{source_table}}) PIVOT ( {{agg_method}} ( {{pivot_column}} ) FOR {{value_column}} IN ( {{list_of_vals}} ) ) as p ( {{dimensions}}, {{list_of_vals_no_quote}})
\ No newline at end of file
+{%- set get_distinct_vals -%}
+    select distinct {{ value_column }}
+    from {{ source_table }}
+    limit 1000
+{%- endset -%}
+{%- set results = run_query(get_distinct_vals) -%}
+{%- set distinct_vals = results[results.columns[0]].to_list() -%}
+
+SELECT {{ dimensions }}, {{ distinct_vals | join(", ") }}
+FROM ( SELECT {{ dimensions }}, {{ pivot_column }}, {{ value_column }} FROM {{ source_table }})
```

Comment:
We probably want to get into the habit of quoting our column names to avoid issues down the road. Something like:

```
"{{ distinct_vals | join('", "') }}"
```

Reply:
Having a special jinja filter would be super cool and probably make writing this stuff much easier.